django_ses/signals.py

```diff
@@ -11,14 +11,45 @@
 click_received = Signal()
 
 
+def _blacklist_recipients(recipients):
+    if len(recipients) == 0:
+        return
+
+    recipients = [email.lower() for email in recipients]
+
+    # If we received only 1 recipients, don't waste extra queries trying to find
+    # out if the recipient has already been blacklisted; just attempt to
+    # blacklist it.
+    if len(recipients) == 1:
+        models.BlacklistedEmail.objects.get_or_create(email=recipients[0])
+        return
+
+    qs = models.BlacklistedEmail.objects.filter(email__in=recipients)
+    blacklisted_emails = set(qs.values_list("email", flat=True))
+    unblacklisted_emails = [email for email in recipients if email not in blacklisted_emails]
+
+    # Try to bulk-insert the unblacklisted emails. If the operation fails (a
+    # possibility only when using Oracle because it doesn't support
+    # ignore_conflicts), fallback to one-by-one insertion.
+    try:
+        models.BlacklistedEmail.objects.bulk_create([
+            models.BlacklistedEmail(email=email) for email in unblacklisted_emails
+        ], ignore_conflicts=True)
+    except Exception:
```

Comment:
I wonder if we can be more specific here and catch only the database error that could occur if there are conflicts.

Reply:
[It looks like](https://github.com/django/django/blob/5ed72087c450af1a5da138bdfa674a069cf3f09c/tests/bulk_create/tests.py#L381) it throws `NotSupportedError`. So I guess we should be good handling that exception!